include/uart.h

```diff
@@ -116,8 +255,24 @@ struct uart_device_config {
 #endif
 };
 
+#endif /* CONFIG_UART_HIGH_LEVEL_API */
+
 /** @brief Driver API structure. */
 struct uart_driver_api {
+
+#ifdef CONFIG_UART_HIGH_LEVEL_API
+
+	int (*tx)(struct device *dev, u8_t *buf, size_t len);
+	int (*rx_en)(struct device *dev, u8_t *buf, size_t len);
+	int (*rx_buf_rsp)(struct device *dev, u8_t *buf, size_t len);
```

Comment:
Why would you need 2 functions (_en and _buf_rsp)? 

One rx function could do both properly, it just requires drivers to take care if cb is present and when it's being called.